tests/pageobjects/IframeAPI.ts

```diff
@@ -0,0 +1,90 @@
+import { Participant } from '../helpers/Participant';
+import { LOG_PREFIX } from '../helpers/browserLogger';
+
+/**
+ * The Iframe API and helpers from iframeAPITest.html
+ */
+export default class IframeAPI {
+    private participant: Participant;
+
+    /**
+     * Initializes for a participant.
+     * @param participant
+     */
+    constructor(participant: Participant) {
+        this.participant = participant;
+    }
+
+    /**
+     * Returns the json object from the iframeAPI helper.
+     * @param event
+     */
+    async getEventResult(event: string): Promise<any> {
+        return this.participant.driver.execute(
+            eventName => {
+                const result = window.jitsiAPI.test[eventName];
+
+                if (!result) {
+                    return false;
+                }
+
+                return result;
+            }, event);
+    }
+
+    /**
+     * Adds an event listener to the iframeAPI.
+     * @param eventName The event name.
+     */
+    async addEventListener(eventName: string) {
+        return this.participant.driver.executeAsync((event, prefix, done) => {
+            console.log(`${new Date().toISOString()} ${prefix} Adding listener for event: ${event}`);
+            window.jitsiAPI.addListener(event, evt => {
```

Comment:
what's with the date?

Reply:
To make the logs look the same way as the js client logs.
```
2024-12-09T23:08:07.278Z [modules/UI/videolayout/LargeVideoManager.js] scheduleLargeVideoUpdate: Remote track RemoteTrack[userID: 23f85c98, type: video, ssrc: 2134074178, p2p: false, sourceName: 23f85c98-v0, status: {readyState: live, muted: false, enabled: true}], isVideoMuted=false, streamingStatusActive=true, isVideoRenderable=true, showAvatar=false
2024-12-09T23:08:09.659Z [MeetTest]  Adding listener for event: participantRoleChanged

```

Reply:
Ah gotcha!